developer-docs-site/docs/concepts/events.md

```diff
@@ -2,8 +2,48 @@
 title: "Events"
 slug: "events"
 ---
+Events are emitted during the execution of a transaction. Each Move module can define its own events and choose when to emit the events upon execution of the module. Aptos Move starts with events with handle on mainnet launch, then switch to module events since v1.7.
 
-Events are emitted during the execution of a transaction. Each Move module can define its own events and choose when to emit the events upon execution of the module. For example, during a [coin transfer](../tutorials/first-transaction.md), both the sender and receiver's accounts will emit `SentEvent` and `ReceivedEvent`, respectively. This data is stored within the ledger and can be queried via the REST interface's [Get events by event handle](https://fullnode.devnet.aptoslabs.com/v1/spec#/operations/get_events_by_event_handle).
+# Module Events
+Since v1.7, Aptos Move released a new event framework, module events. Every event stream is identified by an event struct type instead of a handle.
+To define an event struct, just add attribute `#[event]` to a normal Move struct that has `drop + copy` abilities. For example,
+```rust
+/// 0xcafe::my_module_name
+/// An example module event struct denotes a coin transfer.
+#[event]
+struct TransferEvent<Coin> has copy, drop {
+  sender: address,
+  receiver: address,
+  amount: u64
+}
+```
+Now `TransferEvent<CoinA>` and `TransferEvent<CoinB>` are two different event structs which can be used to emit events so `0x1::event::emit` could be called to emit events.
+for example,
+```rust
+// Define an event.
+let event = TransferEvent<Coin> {
+    sender: 0xcafe,
+    receiver: 0xface,
+    amount: 100
+};
+// Emit the event just defined.
+0x1::event::emit(event);
+```
+The module event has been emitted.
```

Comment:
This needs more details on what it really means. At least:
1. Where did the events get stored?
2. How do devs find them off chain? Link to indexing doc if any
3. Can events be accessed on chain? How about in tests?

Reply:
2 is not ready yet. I will add the other two.